### PR TITLE
Repositories responsabilities correction

### PR DESCRIPTION
**Pull request by Juan-Pablo-Chavez-AS**

### CODE REVIEW

File: src/controllers/client.controller.ts [hunk 1]
```diff
@@ -18,9 +19,14 @@ export default class ClientController {
   public async createClient(req: Request, res: Response) {
     try {
       const clientData = req.body as clientInput;
-      const response = await this.repository.createClient(clientData);
+      const clientDataWithHash = { ...clientData, password_hash: 'whatever' }; // WIP: hashing
```
Comment (josueveliz-assure) [inline src/controllers/client.controller.ts]: Nice use to spread operator but is the comment is unnecessary.

File: src/repositories/client.repository.ts [hunk 2]
```diff
@@ -1,31 +1,27 @@
 import dbClient from '../config/db.config';
 import { clientInput, LoginCredentials, UserIdentity } from '../types/types';
-import JWTManager from '../auth/jwt/jwt.auth';
 import { JsonWebTokenError } from 'jsonwebtoken';
 
 export default class ClientRepository {
-  private static readonly clientInfoSelect = {
+  private static readonly clientBasicInfoSelect = {
```
Comment (josueveliz-assure) [inline src/repositories/client.repository.ts:6]: KUDOS for improving the name variable

**Review by josueveliz-assure — APPROVED** [review]
LGTM! +1

File: src/controllers/client.controller.ts [hunk 3]
```diff
@@ -18,9 +19,14 @@ export default class ClientController {
...
 
-      return res.cookie('jwt', response.token).status(201).json(response);
+      const newClient = await this.repository.createClient(clientDataWithHash);
+
+      const clientToken = JWTManager.generateToken(newClient);
```
Comment (jacovzap-assuresoft) [inline src/controllers/client.controller.ts]: You are breaking single responsability here, abstract the token generation in another function

**Review by jacovzap-assuresoft — COMMENTED** [review]
-1

File: src/controllers/client.controller.ts (hunk 1 above)
Comment (Juan-Pablo-Chavez-AS) [reply src/controllers/client.controller.ts]: Forgot to remove WIP comment, I'll remove it

File: src/controllers/client.controller.ts (hunk 3 above)
Comment (Juan-Pablo-Chavez-AS) [reply src/controllers/client.controller.ts]: In my next commit, tokens are no longer created in the user creation endpoint. thx for the review.

**Review by jacovzap-assuresoft — APPROVED** [review]
LGTM +1

**Pull request merged**